src/devices/Seesaw/SeesawBase.cs

```diff
@@ -0,0 +1,172 @@
+        /// </summary>
+        protected void Initialize(I2cDevice i2cDevice)
+        {
+            I2cDevice = i2cDevice;
+
+            SoftwareReset();
+
+            Task.Delay(10).Wait();
+
+            if(ReadByte(Seesaw_Module.Status, Seesaw_Function.Status_Hw_Id) != Sessaw_Hardware_Id)
+            {
+                throw new NotSupportedException($"The hardware on I2C Bus {I2cDevice.ConnectionSettings.BusId}, Address 0x{I2cDevice.ConnectionSettings.DeviceAddress:X2} does not appear to be an Adafruit SeeSaw module");
+            }
+
+            _options = GetOptions();
+            Version = GetVersion();
+        }
+
+        /// <summary>
+        /// Get the firmware version of the Seesaw board. 
+        /// </summary>
+        /// <returns>Returns the seesaw version.</returns>
+        protected uint GetVersion()
+        {
+            return BinaryPrimitives.ReadUInt32BigEndian(Read(Seesaw_Module.Status, Seesaw_Function.Status_Version, 4));
+        }
+
+        /// <summary>
+        /// Gets the options/modules present on the Seesaw board
+        /// </summary>
+        /// <returns>Returns 32 bit integer where the bits represent the options/modules present on the Seesaw board.</returns>
+        protected uint GetOptions()
+        {
+            return BinaryPrimitives.ReadUInt32BigEndian(Read(Seesaw_Module.Status, Seesaw_Function.Status_Options, 4));
+        }
+
+        /// <summary>
+        /// Write a byte to the I2cDevice connected to the Seesaw board. 
+        /// </summary>
+        /// <param name="moduleAddress">An Seesaw_Module enum that represents the module that we are writing to.</param>
+        /// <param name="functionAddress">An Seesaw_Function enum that represents the Seesaw function to be called.</param>
+        /// <param name="value">The byte value ro be send as a parameter to the Seesaw device.</param>
+        protected void WriteByte(Seesaw_Module moduleAddress, Seesaw_Function functionAddress, byte value)
+        {
+            Write(moduleAddress, functionAddress, new byte[] { value });
+        }
+
+        /// <summary>
+        /// Write a series of bytes to the I2cDevice connected to the Seesaw board. 
+        /// </summary>
+        /// <param name="moduleAddress">An Seesaw_Module enum that represents the module that we are writing to.</param>
+        /// <param name="functionAddress">An Seesaw_Function enum that represents the Seesaw function to be called.</param>
+        /// <param name="data">The Span containg data to be send as a parameter or parameters to the Seesaw device.</param>
+        protected void Write(Seesaw_Module moduleAddress, Seesaw_Function functionAddress, Span<byte> data)
+        {
+            const int StackThreshold = 512;
+
+            Span<byte> buffer = data.Length < StackThreshold ? stackalloc byte[data.Length + 2] : new byte[data.Length + 2];
+
+            buffer[0] = (byte) moduleAddress;
```

Comment:
nit: no space between cast operator and operand

Reply:
(same below)

Reply:
Yes - for some reason Visual studio wasn't applying the .editorconfig suggestions. Hopefully I have found all occurrences and corrected in the latest PR update